### ISSUE TITLE
FlowData.write_fcs attempts to write strings to file opened in binary mode

### ISSUE THREAD

**Issue by lgrozinger** [whitews/FlowIO#4]
#### Issue:
`FlowData.write_fcs` will fail in Python3

#### Reproduce:

Running the following under Python3.X:
```
    data = flowio.FlowData(fcsfile)
    data.write_fcs(fcsfile, extra=annotations)
```

will result in:
```
  File "/home/campus.ncl.ac.uk/b8051106/.local/lib/python3.8/site-packages/flowio/flowdata.py", line 396, in write_fcs
    fh.write('FCS3.1')
TypeError: a bytes-like object is required, not 'str'
```

**Comment by lgrozinger** [comment]
Between Python2.7 and Python3.X, native strings [changed](https://portingguide.readthedocs.io/en/latest/strings.html) from bytes to text.

`FlowData.write_fcs` opens the FCS file in binary write mode, and writes a bunch of literal strings (e.g.         `fh.write('FCS3.1')`), which means the code works in Python2 but breaks in Python3.

The suggestion is to change all the literals (and strings obtained otherwise) to explicitly use the `bytes` type. That is, to change e.g.  `fh.write('FCS3.1')` to  `fh.write(b'FCS3.1')`. The two strings have the same type in Python2 (`str`), so this should still work as before. In Python3 the `b'FCS3.1'` has the type of `bytes` as required.

**Comment by whitews** [comment]
Hi Lewis,

Thanks for taking the time to report this issue. You are correct that the behavior changed in Python 3 and the code in `create_fcs` was written back in the Python 2 days. 

Odd that I haven't run into this error yet as FlowKit uses this function and is tested against Python 3. I wonder if this is OS dependent (I mainly use Linux). What OS are you using? I'd like to write a test case for this.

Thanks,
Scott

**Comment by lgrozinger** [comment]
> Thanks for taking the time to report this issue.

No problem, thanks for replying, and for authoring this very useful package.

> Odd that I haven't run into this error yet as FlowKit uses this function and is tested against Python 3. I wonder if this is OS dependent (I mainly use Linux). What OS are you using? I'd like to write a test case for this.

I see. I think the confusion here is that I came across `FlowData.write_fcs` before `create_fcs`.


I expected that [`write_fcs`](https://github.com/whitews/FlowIO/blob/abeb1934e5fa8278713df02bcb5020f7935ca872/flowio/flowdata.py#L395) would do what `create_fcs` does, but with a `FlowData` object instead of explicitly passing in event data etc. However, although `write_fcs` does look like it does this upon first glance, further inspection reveals that even if the `bytes`/`str` bug was fixed, the method does not write the event data correctly anyway.

Anyway, `create_fcs` does what I want, apart from I have to pass the `event_data`, `channel_names`, etc. explicitly. In summary,
I think one of two things could happen:

1. Delete the `write_fcs` method, it doesn't work anyway, and it confused me ;)
2. Keep `write_fcs` but just as a wrapper for `create_fcs`, so I can do:
```
    data = flowio.FlowData(fcsfile)
    data.write_fcs(fcsfile, extra=annotations)
```

I prefer option 2.
